lib/ansible/modules/network/avi/avi_virtualservice.py

```diff
@@ -0,0 +1,451 @@
+#!/usr/bin/python
+#
+# Created on Aug 25, 2016
+# @author: Gaurav Rastogi (grastogi@avinetworks.com)
+#          Eric Anderson (eanderson@avinetworks.com)
+# module_check: not supported
+# Avi Version: 16.3.6
+#
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'], 'supported_by': 'community', 'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: avi_virtualservice
+author: Gaurav Rastogi (grastogi@avinetworks.com)
+
+short_description: Module for setup of VirtualService Avi RESTful Object
+description:
+    - This module is used to configure VirtualService object
+    - more examples at U(https://github.com/avinetworks/devops)
+requirements: [ avisdk ]
+version_added: 2.3
+options:
+    state:
+        description:
+            - The state that should be applied on the entity.
+        required: false
```

Comment:
If a parameter is not required, you do not have to add `required: false` as that is implicit.
Is is easier to find what is required if these are not listed.